pyparsing/results.py

```diff
@@ -66,41 +66,46 @@ def test(s, fn=repr):
         - year: 1999
     """
 
+    __slots__ = [
+        "_name",
+        "_parent",
+        "_all_names",
+        "_modal",
+        "_list",
+        "_dict",
+        "__weakref__",
+    ]
+
     def __new__(cls, toklist=None, name=None, asList=True, modal=True):
         if isinstance(toklist, ParseResults):
             return toklist
-        retobj = object.__new__(cls)
-        retobj.__doinit = True
-        return retobj
+        self = object.__new__(cls)
+        self._name = None
+        self._parent = None
+        self._all_names = set()
+        self._modal = modal
+        if toklist is None:
+            toklist = []
+        if isinstance(toklist, list):
+            self._list = toklist[:]
+        elif isinstance(toklist, _generator_type):
+            self._list = list(toklist)
+        else:
+            self._list = [toklist]
+        self._dict = dict()
+        return self
 
     # Performance tuning: we construct a *lot* of these, so keep this
     # constructor as small and fast as possible
     def __init__(
         self, toklist=None, name=None, asList=True, modal=True, isinstance=isinstance
     ):
-        if self.__doinit:
```

Comment:
Please change `_result_xxx` names to just `_xxx`
Leave `_toklist` and `_tokdict`. Chopping to `_list` and `_dict` are not sufficiently descriptive (or expand to `_token_list` and `_token_name_dict` or `_results_name_dict` for more self-descriptive names)